server/market/bookrouter.go

```diff
@@ -536,6 +537,23 @@ func (r *BookRouter) sendNote(route string, subs *subscribers, note interface{})
 	}
 }
 
+// cancelOrderToMsgOrder converts an *order.CancelOrder to a
+// *msgjson.BookOrderNote.
+func cancelOrderToMsgOrder(o *order.CancelOrder, mkt string) *msgjson.BookOrderNote {
```

Comment:
This is just a question, does this return a pointer because it's more idiomatic, or is there another reason? I ask because the value is what is used.

Reply:
Passing large structs by value is generally not advisable.  `msgjson.BookOrderNote` is pretty large.  Although the consumer copies it, I didn't want another copy with the return value.  The compiler might elide any of these copies I suppose, but I'd rather write such functions that don't make you get a copy of a large struct.